crates/types/src/extension.rs

```diff
@@ -51,6 +51,10 @@ impl packed::RawL2Transaction {
     pub fn hash(&self) -> [u8; 32] {
         self.as_reader().hash()
     }
+
+    pub fn is_chain_id_protected(&self) -> bool {
```

Comment:
```suggestion
    pub fn is_eip155_tx(&self) -> bool {
```

I rename this function because I think "protected chain id" is difficult to understand, and "eip155" points to more useful information.

Reply:
The `chain_id` should be interpreted by different EOA locks, since we only have the ETH EOA, it is correct to call it EIP-155. But actually the tx isn't encoded in EIP-155, it encoded in Godwoken L2TX format, so I prefer the original name.

Reply:
Now LGTM. The core concept is "protect tx by chain id". my intention is to reduce the barrier of understanding by using the same concept.